src/idl_parser.cpp

```diff
@@ -2004,18 +2048,39 @@ CheckedError Parser::DoParse(const char *source, const char **include_paths,
   while (token_ != kTokenEof) {
     if (opts.proto_mode) {
       ECHECK(ParseProtoDecl());
-    } else if (token_ == kTokenNameSpace) {
-      ECHECK(ParseNamespace());
-    } else if (token_ == '{') {
+  }
+  else if (token_ == kTokenNameSpace) {
+    ECHECK(ParseNamespace());
+
+  } else if (token_ == '{' || token_ == '[') {
```

Comment:
That '[' there is to support the case where the top level element in the json file is a vector, which is valid json.
Example taken from the TopLevelListTest in the PR:
`[ {a:10, b:20}, {...} ]  `

Corresponding FBS could be:
`struct Test { a:short; b:byte; } table T { F:[Test]; } root_type T;
`

I could not find a way to only change ParseTable for that case, as ParseTable relies on being called down the callstack from a ParseVector() call.